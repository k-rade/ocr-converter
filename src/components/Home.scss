.home {


    &-title {
        margin: 0;
        font-size: 3rem;
        font-weight: 800;
        color: var(--color-white-two);
        text-align: center;
    }

    &-description {
        margin: 0 0 2rem;
        color: var(--color-gray-900);
        font-size: 1.125rem;
        text-align: center;
    }
}

.converters {
    display: flex;
    flex-wrap: wrap;
    gap: 1.5rem;
}

.converter-card {
    flex: 1;
    padding: 1.5rem;
    min-width: 16rem;
    height: 16rem;
    width: 100%;
    background-color: var(--color-gray-200);
    text-decoration: none;
    border-radius: 2px;
    position: relative;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    overflow: hidden;

    .icon-button svg {
        fill: none;
    }

    &:hover {
        .abstract-line>path {
            stroke: var(--color-primary-500);
        }

        .icon-button {
            border-color: var(--color-primary-500);

            svg>path {
                stroke: var(--color-primary-500) !important;
            }
        }
    }

    .abstract-line {
        height: 100%;
        top: 0;
        right: 0;
        position: absolute;

        &-one {
            top: -4px;
            right: -4px;
        }

        &-two {
            right: -6px;
        }
    }

    h2 {
        white-space: nowrap;
        color: var(--color-white-two);
        margin: 0;
    }

    p {
        margin: 1rem 0 0;
        color: var(--color-gray-900);
    }
}