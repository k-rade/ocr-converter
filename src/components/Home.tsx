import { NavLink } from "react-router-dom";
import { LANGUAGES } from "../constants/languages";
import IconButton from "./IconButton";
import { ReactComponent as AbstractLinesOne } from "../icons/abstract-line-one.svg";
import { ReactComponent as AbstractLinesTwo } from "../icons/abstract-line-two.svg";
import { ReactComponent as ArrowRightIcon } from "../icons/arrow-right.svg";

import "./Home.scss";

const Home = (): JSX.Element => (
  <div className="home">
    <h2 className="home-title">Convertify</h2>
    <p className="home-description">Set of simple converter tools</p>

    <div className="converters">
      <NavLink to="/ocr-converter" className="converter-card">
        <IconButton icon={<ArrowRightIcon />} />
        <AbstractLinesOne />
        <div>
          <h2>Image to text</h2>
          <p>
            Extracts text from images. <br /> Supports {LANGUAGES.length}{" "}
            languages.
          </p>
        </div>
      </NavLink>

      <NavLink to="/cyrilic-to-latin" className="converter-card">
        <IconButton icon={<ArrowRightIcon />} />
        <AbstractLinesTwo />
        <div>
          <h2>Cyrilic to latin</h2>
          <p>
            Converts cyrilic letters <br /> into latin letters.
          </p>
        </div>
      </NavLink>
    </div>
  </div>
);

export default Home;
