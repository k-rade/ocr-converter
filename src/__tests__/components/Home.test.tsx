import { render } from "@testing-library/react";
import { MemoryRouter, useLocation } from "react-router-dom";
import Home from "../../components/Home";

jest.mock("react-router-dom", (): any => {
  const actualImplementation = jest.requireActual("react-router-dom");
  return {
    ...actualImplementation,
    useLocation: jest.fn(),
  };
});

describe("Home", (): void => {
  beforeEach((): void => {
    (useLocation as jest.Mock).mockImplementation(() => ({
      pathname: "/test-route",
    }));
  });

  it("should render", (): void => {
    const TestComponent = (
      <MemoryRouter>
        <Home />
      </MemoryRouter>
    );
    const { baseElement } = render(TestComponent);

    expect(baseElement).toMatchSnapshot();
  });
});
