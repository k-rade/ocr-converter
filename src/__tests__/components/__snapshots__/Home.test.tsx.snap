// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render 1`] = `
<body>
  <div>
    <div
      class="home"
    >
      <h2
        class="home-title"
      >
        Convertify
      </h2>
      <p
        class="home-description"
      >
        Set of simple converter tools
      </p>
      <div
        class="converters"
      >
        <a
          class="converter-card"
          href="/ocr-converter"
        >
          <div
            class="icon-button-container"
          >
            <button
              class="icon-button"
              type="button"
            >
              <svg>
                arrow-right.svg
              </svg>
            </button>
          </div>
          <svg>
            abstract-line-one.svg
          </svg>
          <div>
            <h2>
              Image to text
            </h2>
            <p>
              Extracts text from images. 
              <br />
               Supports 
              67
               
              languages.
            </p>
          </div>
        </a>
        <a
          class="converter-card"
          href="/cyrilic-to-latin"
        >
          <div
            class="icon-button-container"
          >
            <button
              class="icon-button"
              type="button"
            >
              <svg>
                arrow-right.svg
              </svg>
            </button>
          </div>
          <svg>
            abstract-line-two.svg
          </svg>
          <div>
            <h2>
              Cyrilic to latin
            </h2>
            <p>
              Converts cyrilic letters 
              <br />
               into latin letters.
            </p>
          </div>
        </a>
      </div>
    </div>
  </div>
</body>
`;
